stdlib/public/core/Sort.swift

```diff
@@ -506,9 +510,13 @@ internal func _findNextRun<C: RandomAccessCollection>(
 }
 
 extension UnsafeMutableBufferPointer {
+  // FIXME(ABI): unused return value
   /// Merges the elements at `runs[i]` and `runs[i - 1]`, using `buffer` as
   /// out-of-place storage.
   ///
+  /// The unused return value is legacy ABI. It was originally added as a
+  /// workaround for a compiler bug (now fixed). See SR-14750 (rdar://45044610).
```

Comment:
Perhaps, for the user who stumbles on this documentation of this (public, underscored) method and other similar ones, one might consider stating explicitly that the return value is an arbitrary value. I can imagine that "unused" could be confusing since it's up to the caller whether it's unused or not (and "legacy ABI" may not mean much to many folks in terms of clarifying this)--in this case, we're merely trying to say that the value is of no use.

I wonder if it'd be most fitting to write this information explicitly in a `- Returns: ` section; it isn't necessary to understand what the function is doing.

Reply:
Huh -- perhaps we should also change this method to be `@inlinable internal`. It doesn't seem like a good idea to expose internal implementation details as public entry points.

(BTW, I'm not sure why we ended up using an `Array` here for `runs` -- it'd be too painful to change it now, but I hope it gets stack allocated...)

Reply:
Making them internal now is a good idea. I think that when they were added internal was still ill-advised for the standard library, though now it is fine. We can also further document the expectations with tests.